crates/router/src/connector/billwerk/transformers.rs

```diff
@@ -136,28 +256,68 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
+        let error_response = if item.response.error.is_some() || item.response.error_state.is_some()
+        {
+            Some(types::ErrorResponse {
+                code: item
+                    .response
+                    .error_state
+                    .clone()
+                    .unwrap_or(consts::NO_ERROR_CODE.to_string()),
+                message: item
+                    .response
+                    .error_state
+                    .unwrap_or(consts::NO_ERROR_MESSAGE.to_string()),
+                reason: item.response.error,
+                status_code: item.http_code,
+                attempt_status: None,
+                connector_transaction_id: Some(item.response.handle.clone()),
+            })
+        } else {
+            None
+        };
+        let payments_response = types::PaymentsResponseData::TransactionResponse {
+            resource_id: types::ResponseId::ConnectorTransactionId(item.response.handle.clone()),
+            redirection_data: None,
+            mandate_reference: None,
+            connector_metadata: Some(serde_json::json!({
```

Comment:
Why this has to be persisted in db? do we need this in further calls?

Reply:
Currently their is no requirement as of now, this is just a fallback if such use case comes

Reply:
Please remove this, as we have connector_transaction_id as reference anyways